### PR TITLE
Remove MD5 usage and dependency from UI

### PR DESCRIPTION
**Pull request by kialam**
##### SUMMARY

Built off of #2718

Related ansible/tower#814

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - UI

##### AWX VERSION

```
awx: 2.1.0
```

### CODE REVIEW

**Review by ryanpetrello — APPROVED** [review]
😄

**Comment by mabashian** [comment]
Code makes sense and looks good.  Seems like we need to discuss what the expectation is for the package-lock file as a lot's changed in there.  Also, not sure why the api tests would be failing with this PR 🤷‍♂️

**Comment by wenottingham** [comment]
> Also, not sure why the api tests would be failing with this PR 

The license checker runs as an 'api' test as it was the most convenient place to shove it.

**Comment by kialam** [comment]
> > Also, not sure why the api tests would be failing with this PR
> 
> The license checker runs as an 'api' test as it was the most convenient place to shove it.

@wenottingham does the license checker also look in the /ui directory? I found a second `angular-md5` license there.

**Comment by wenottingham** [comment]
Yes, it should be.

**Comment by Spredzy** [comment]
recheck

**Comment by ryanpetrello** [comment]
recheck

File: awx/ui/client/src/templates/job_templates/factories/hash-setup.factory.js [hunk 1]
```diff
@@ -9,10 +9,12 @@ export default
...
             master[check_field] = default_val;
 
-            scope.genMD5 = function (fld) {
-                var now = new Date();
-                scope[fld] = md5.createHash('AnsibleWorks' + now.getTime());
-                scope.$emit('NewMD5Generated');
+            // Original gist here: https://gist.github.com/jed/982883
+            scope.genHash = function (fld) {
+                scope[fld] = ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
+                    (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
```
Comment (ryanpetrello) [inline awx/ui/client/src/templates/job_templates/factories/hash-setup.factory.js:15]: https://caniuse.com/#feat=getrandomvalues

**Review by jlmitch5 — APPROVED** [review]
all the changes from MD5 to hash look good.  the UUID generation looks okay I think?  so far as I understand it

**Comment by ryanpetrello** [comment]
From a testing perspective, the thing I mostly am concerned about coverage-wise is ensuring that `crypto.getRandomValues` _does_ seem to work across our supported browser matrix (
https://caniuse.com/#feat=getrandomvalues suggests that it does).

**Review by jakemcdermott — APPROVED** [review]
verified locally:

hostconfigkey1

File: awx/ui/package-lock.json [hunk 2]
```diff
@@ -154,16 +154,14 @@
...
     },
     "angular-codemirror": {
       "version": "git+https://git@github.com/ansible/angular-codemirror.git#447f071eff8f6fde7b5ec769c57c7dc98a014fdf",
-      "from": "git+https://git@github.com/ansible/angular-codemirror.git#v1.1.2",
       "requires": {
-        "angular": "~1.6.6",
-        "codemirror": "^5.17.0",
-        "jquery": "^3.2.1"
+        "angular": "1.6.10",
+        "codemirror": "5.38.0",
```
Comment (jakemcdermott) [inline awx/ui/package-lock.json]: I'm unclear on why this is changing?

File: awx/ui/package-lock.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
       "integrity": "sha1-w8p0NJOGSMPg2cHjKN1otiLChMo=",
```
Comment (jakemcdermott) [inline awx/ui/package-lock.json]: I think all we need here is:

```
make clean
npm uninstall --prefix awx/ui --save angular-md5
```

**Review by jakemcdermott — CHANGES_REQUESTED** [review]
This mostly looks good to me - just one change (see comments) and I think this should be good to go!

Happy to verify again locally with more browsers when ready 👍

**Review by jakemcdermott — APPROVED** [review]
code changes now look good to me! 👍 thank you @kialam 

I'll verify again with a supported firefox version shortly and then I think this should be ready to go!

File: awx/ui/README.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 ## Requirements
 - node.js 8.x LTS
-- npm 5.x LTS
+- npm >=5.10
```
Comment (jakemcdermott) [inline awx/ui/README.md:5]: 👍

**Comment by unlikelyzero** [comment]
@ryanpetrello @jakemcdermott what will happen with previously generated values if a user upgrades?

**Comment by jakemcdermott** [comment]
@unlikelyzero 
Nothing, I'd presume? In the end of the day the hash is just a string. Old values already persisted in the db and sent over from the API should still be ~displayable.~

**Comment by jakemcdermott** [comment]
still looks good to me on ffox esr:

hostconfigkey5

**Comment by jakemcdermott** [comment]
e2e, etc. still checks out. Probably ready to merge this finally?

**Comment by jakemcdermott** [comment]
update: probably should get wf convergence in first, as they both update `package.json`.

**Comment by ryanpetrello** [comment]
Agreed on holding off the merge until after WF convergence is in.

@unlikelyzero nothing will happen w/ the old values; this code is only used to generate a random string that is passed to and stored in the API.

**Comment by jakemcdermott** [comment]
- reverified on firefox esr
- e2e still looks good

**Pull request merged**